### PR TITLE
Add test 'Check links in header page FO'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.7.x
| Description?      | Add test 'Check links in header page FO'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      |  TEST_PATH="functional/FO/02_headerAndFooter/01_checkLinksInHeader.js" URL_FO=yourShopURL npm run specific-test
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23503)

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    await testContext.addContextItem(this, 'testIdentifier', 'checkContactUsHeaderLink', baseContext);
+
+    // Check Contact us
+    await homePage.goToHeaderLink(page, 'Contact us');
+
+    const pageTitle = await contactUsPage.getPageTitle(page);
+    await expect(pageTitle, 'Fail to open FO login page').to.contains(contactUsPage.pageTitle);
+  });
+
+  it('check languages link', async function () {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js]:
```suggestion
  it('should check languages link', async function () {
```

File: tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  });
+
+  it('should check search link', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'checkSearchLink', baseContext);
+
+    await homePage.searchProduct(page, Products.demo_8.name);
+
+    const pageTitle = await searchResultsPage.getPageTitle(page);
+    await expect(pageTitle).to.equal(searchResultsPage.pageTitle);
+  });
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js]: Maybe this part should be on a different test, wdyt ?

File: tests/UI/pages/FO/FObasePage.js [hunk 3]
```diff
@@ -68,24 +70,56 @@ module.exports = class FOBasePage extends CommonPage {
...
+        break;
+
+      case 'Logo':
+        selector = this.desktopLogo;
+        break;
+
+      default:
+        throw new Error(`The page ${link} was not found`);
+    }
+    return this.clickAndWaitForNavigation(page, selector);
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/FObasePage.js]:
```suggestion
    }

    return this.clickAndWaitForNavigation(page, selector);
```

File: tests/UI/pages/FO/FObasePage.js [hunk 4]
```diff
@@ -68,24 +70,56 @@ module.exports = class FOBasePage extends CommonPage {
...
   }
 
+  // Header methods
+  /**
+   * Go to header link
+   * @param page
+   * @param link
+   * @returns {Promise<void>}
+   */
+  async goToHeaderLink(page, link) {
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/FObasePage.js]: Maybe naming the function `clickOnHeaderLink` instead. WDYT ?

File: tests/UI/pages/FO/FObasePage.js [hunk 5]
```diff
@@ -208,7 +242,19 @@ module.exports = class FOBasePage extends CommonPage {
...
+  /**
+   * Search product
+   * @param page
+   * @param productName
+   * @returns {Promise<void>}
+   */
+  async searchProduct(page, productName) {
+    await this.setValue(page, this.searchInput, productName);
+    await page.keyboard.press('Enter');
+    await page.waitForNavigation();
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/FObasePage.js]: Please set the navigation method here 'networkidle' or 'load' or 'domcontentloaded'

File: tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js]: I just automated the steps of exploratory tests on test rail :
Check all the links in the header:
- contact us
- sign in
- logo
- all the menu and sub-menu links

Comment (boubkerbribri) [reply tests/UI/campaigns/functional/FO/02_headerAndFooter/01_checkLinksInHeader.js]: could be seperated on testrail too

File: tests/UI/pages/FO/FObasePage.js (hunk 5 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/FObasePage.js]: Done thx :+1:

**Review by boubkerbribri — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**